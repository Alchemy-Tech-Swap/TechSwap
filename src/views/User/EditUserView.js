import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import EditUserComp from '../../components/Users/EditUserComp';
import { editUserDetails, getUserById } from '../../services/users';

export default function EditUserView({ currentUser }) {
  const [user, setUser] = useState({ name: '', slack_id: '', linkedin_url: '' });
  const [alert, setAlert] = useState('');
  const history = useHistory();
  const [loading, setLoading] = useState(true);
  const id = currentUser.user.id;

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getUserById(id);
      setUser(resp);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  const onChange = ({ target }) => {
    setUser((prevState) => ({ ...prevState, [target.name]: target.value }));
  };

  const onSubmit = async (e) => {
    e.preventDefault();

    try {
      await editUserDetails(user.id, {
        name: user.name,
        slack_id: user.slack_id,
        linkedin_url: user.linkedin_url,
      });
      history.push(`/profile/${user.id}`);
    } catch (err) {
      setAlert(err.message ? err.message : 'WRONG');
    }
  };

  if (loading) return <h1>LOADING</h1>;

  return (
    <div>
      <p>{alert}</p>
      <EditUserComp
        label="Edit User Info"
        name={user.name}
        slack_id={user.slack_id}
        linkedin_url={user.linkedin_url}
        onChange={onChange}
        onSubmit={onSubmit}
      />
    </div>
  );
}
