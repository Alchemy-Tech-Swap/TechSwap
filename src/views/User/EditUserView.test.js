import EditUserView from './EditUserView';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

test('EditUserView.js should do something', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/profiles/26/edit']}>
      <Route path="/profiles/:id/edit">
        <EditUserView
          currentUser={{
            user: {
              id: '0fcdf7e2-48e8-4f4d-b4e6-5453ce7ee18d',
              created_at: '2022-01-20T23:03:53.531704+00:00',
              email: 'rewq@ewq.vfdsa',
              name: 'rewqrewq',
              slack_id: 'rewqrewq',
              linkedin_url: 'rewqrewq',
              user_id: '0fcdf7e2-48e8-4f4d-b4e6-5453ce7ee18d',
            },
          }}
        />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Edit User Info');

  expect(container).toMatchSnapshot();
});
