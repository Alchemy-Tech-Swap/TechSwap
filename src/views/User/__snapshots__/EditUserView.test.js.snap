// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserView.js should do something 1`] = `
<div>
  <div>
    <p />
    <div>
      <div>
        EDIT
      </div>
      <div>
        Edit User Info
      </div>
      <form>
        <label
          for="name"
        >
          Name: 
        </label>
        <input
          id="name"
          name="name"
          type="text"
          value="rewqrewq"
        />
        <br />
        <label
          for="slack_id"
        >
          Slack: 
        </label>
        <input
          id="slack_id"
          name="slack_id"
          type="text"
          value="rewqrewq"
        />
        <br />
        <label
          for="linkedin_url"
        >
          LinkedIn: 
        </label>
        <input
          id="linkedin_url"
          name="linkedin_url"
          type="text"
          value="rewqrewq"
        />
        <button>
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
